pkg/function/data_test.go

```diff
@@ -535,3 +556,63 @@ func (s *DataSuite) initPVCTemplateParams(c *C, pvc *v1.PersistentVolumeClaim, o
 	tp.Profile = s.profile
 	return tp
 }
+
+func (s *DataSuite) TestBackupWrongPassword(c *C) {
+	tp, _ := s.getTemplateParamsAndPVCName(c, 1)
+
+	// Test backup
+	bp := *newBackupDataBlueprint()
+	bp.Actions["backup"].Phases[0].Args[BackupDataEncryptionKeyArg] = restic.GeneratePassword()
```

Comment:
@DeepikaDixit - can you verify there's a test that does backup/restore/delete on a repository with a custom password?

Reply:
There is a test that does that in this file itself.

I’ll add one for this workflow as well, where Init was old password and later password changed.